shapely/geometry/point.py

```diff
@@ -99,6 +100,14 @@ def z(self):
             raise DimensionError("This point has no z coordinate.")
         return shapely.get_z(self)
 
+    @property
+    @requires_geos("3.12.0")
```

Comment:
I think you can leave out the decorator here, because calling `shapely.has_m` will already raise an error for that

Reply:
Sure. Also I've added a "versionadded" note for the property that shows up in the [docs](https://shapely--2019.org.readthedocs.build/en/2019/reference/shapely.Point.html#shapely.Point.m).